lib/segment/src/index/field_index/geo_index.rs

```diff
@@ -467,6 +472,11 @@ impl GeoMapIndex {
     }
 
     pub fn match_cardinality(&self, values: &[GeoHash]) -> CardinalityEstimation {
+        let max_values_per_point = self.max_values_per_point();
+        if max_values_per_point == 0 {
+            return CardinalityEstimation::exact(0);
+        }
```

Comment:
Not 100% sure if we can shortcut here. It would be awesome if someone else could confirm this.

Reply:
Looks reasonable. if there is no any value in index, cardinality is exact zero.